roboteam_ai/src/world/Robot.cpp

```diff
@@ -141,22 +141,28 @@ unsigned char Robot::getDribblerState() const {
 void Robot::setDribblerState(unsigned char dribbler) {
 
     if (dribbler < 0 || dribbler > 31) {
-        std::cout << "setting invalid dribbler state (" << (int)dribbler <<
+        std::cout << "setting invalid dribbler state (" << (int) dribbler <<
                   ") for robot with id " << id << std::endl;
     }
     else if (! workingDribbler) {
-        std::cout << "setting dribbler state (" << (int)dribbler <<
+        std::cout << "setting dribbler state (" << (int) dribbler <<
                   ") for robot without working dribbler with id " << id << std::endl;
     }
     else {
-        previousDribblerState = dribblerState;
+        if (previousDribblerState != dribblerState) {
+            auto maxStatesChanged = static_cast<int>(
+                    (world::world->getTime() - timeDribblerChanged)/timeToChangeOneDribblerLevel);
+            auto statesChanged = std::max(abs(previousDribblerState - dribblerState), maxStatesChanged);
+            previousDribblerState = previousDribblerState +
+                    abs(previousDribblerState - dribblerState)/(previousDribblerState - dribblerState)*statesChanged;
+        }
         dribblerState = dribbler;
         timeDribblerChanged = world::world->getTime();
     }
 }
 
 bool Robot::isDribblerReady() const {
```

Comment:
Still don't really like this functionality since it is about the ball and not about the robot. The dribbler is already still, it's just the ball that has residual spin.